src/town_cmd.cpp

```diff
@@ -3559,3 +3560,25 @@ void ResetHouses()
 	/* Reset any overrides that have been set. */
 	_house_mngr.ResetOverride();
 }
+
+/* ToDo: documentation */
+void ToggleTownZoneDisplay(Town *town)
+{
+	bool new_show_state = !town->show_zone;
+	town->show_zone = new_show_state;
+
+	int max_dist = _settings_game.economy.dist_local_authority;
+	int max_y;
+
+	for (int dx = -max_dist; dx <= max_dist; dx++) {
+		max_y = Delta(max_dist, dx);
+		for (int dy = -max_y; dy <= max_y; dy++) {
+			TileIndex t = TileAddWrap(town->xy, dx, dy);
+			if (t != INVALID_TILE) {
+				/* ToDo: tidy up _me[t].m8 sets at other places to avoid overwrite */
+				/* ToDo: set _me[t].m8 bit to 0 on load, or load zone toggle status as well */
+				if (ClosestTownFromTile(t, max_dist) == town) SB(_me[t].m8, 15, 1, new_show_state);
```

Comment:
Direct map access without accessors is a no-no.

Reply:
And this is updating the map array from GUI code. This is also not allowed.

Reply:
Thanks for the comment, I will take it into account if this piece of information gets to be stored in me.m8.
As far as I know putting a purely visual thing which can be calculated into the map array is also a no. As the data does not belong there, I used the most direct method to see if this feature deemed to worth a new map sized cache or not. Of course I think I will need some help in the appropriate placement and access of the cache variable if it comes to that.

Reply:
Now I have an accessor.
If this info stays in me.m8, I have to set it through DoCommand?
If so, does this information get synchronized between clients in a multiplayer game?